### DISCUSSION TITLE
barssince

### DISCUSSION THREAD

**Discussion by Hawar-Dzaee** [kernc/backtesting.py#591] [Q&A]
How can I refer to a previous "barssince". 
I know that "barssince" count the candles since the condition was "last time true.

I want to enter a buying order after there have been three consecutive green candles ( twice), as indicated below. 

![Screenshot_1](https://user-images.githubusercontent.com/96496172/153268340-addd7e9d-a3ce-4495-984e-0ee0c109f003.png)

if I set 

entry_condtion = barrsince( self.data.close[-1]>self.data.close[-2]>self.data.close[-3])

if entry_condition>0:
      `self.buy()

it will open a position at the first instance. I want to open on the second instance. 
so my strategy is to (this pattern happens twice) then open a position. and so on.

If this can not be done using "barssince" is there another way to do that?? 
is there "valuewhen"?

**Comment by gypsyzz** [comment]
so you would want to have a second flag that says how many times this entry_condition happened?

`entry_condtion = barrsince( self.data.close[-1]>self.data.close[-2]>self.data.close[-3])
flag_count = 0`

```
if entry_condition>0:
    flag_count = flag_count + 1
    if flag_count%2==0:
        self.buy()
```
